playkit/src/main/java/com/kaltura/playkit/plugins/ads/ima/IMAExoPlugin.java

```diff
@@ -1102,4 +1102,11 @@ public void onBufferEnd() {
             pause();
         }
     }
+
+    @Override
+    public void onSourceError(Exception exoPlayerException) {
```

Comment:
here we should send ad error like it was comming from IMA

sendError(errorType, errorMessage, adException);

Reply:
Added new params in the function and pushed the code.

Reply:
need to add sendError(PKAdErrorType.VIDEO_PLAY_ERROR, errorMsg, exoPlayerException); in the method not the api params